### ISSUE TITLE
BioSQL tests should not require CREATE DATABASE permissions

### ISSUE THREAD

**Issue by peterjc** [biopython/biopython#825]
Ideally from a security point of view, we'd do something like this (e.g. with PostgreSQL):

``` bash
$ psql -U postgres -c "CREATE USER biosql WITH PASSWORD 'testing'"
CREATE ROLE
$ psql -U postgres -c "CREATE DATABASE biosql_test"
CREATE DATABASE
$ psql -U postgres -c "GRANT ALL PRIVILEGES ON DATABASE biosql_test TO biosql"
GRANT
```

And allow local access to this specific database via the `pg_hba.conf` file:

```
# Access from local host for running BioSQL tests                                                                                       
local   "biosql_testing" biosql                            md5                                                                          
host    "biosql_testing" biosql      127.0.0.1/32          md5                                                                          
host    "biosql_testing" biosql      ::1/128               md5 
```

Here I'm creating the empty database `biosql_test` and an associated locked down user account `biosql` with a password `testing` (settings which would then go into the `test_BioSQL_psycopg2.py` file, and be mimicked for TravisCI). I do _not_ want to have to grant the `biosql` user permission to create new databases:

```
$ psql -U postgres -c "CREATE USER biosql WITH PASSWORD 'testing' CREATEDB"
CREATE ROLE
```

Currently (aside from SQLite), `common_BioSQL.py` function `_do_db_create()` essentially runs `DROP DATABASE biosql_test` and then `CREATE DATABASE biosql_test` before (re)importing the schema. In theory we could drop any tables and then (re)import the schema instead?

**Comment by emily-bradley** [comment]
@peterjc I would like to work on this issue.

**Comment by peterjc** [comment]
I am the de facto maintainer for the Biopython BioSQL code, but I am by no means a database expert. So, if you know that side of things this is a nice choice to work on, albeit a relatively low priority issue.

Note the way we configure the usernames and passwords for the BioSQL tests has changed since I opened this issue - see file ``Tests/biosql.ini.sample`` and how it is used in ``.travis.yml`` for our continuous integration tests.

https://github.com/biopython/biopython/blob/master/Tests/biosql.ini.sample

However, the general principle remains true, and would apply in cases like wanting to run the tests on a relatively locked down machine.

**Comment by peterjc** [comment]
Implemented for PostgreSQL in #1420.
